### PR TITLE
MAINT: span model enhancement

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR enhances the Span Event model in the following aspects:
* add serviceName attribute
* add equal method in DefaultTraceGroupFields
* add fromSpan API in JacksonSpan
 
### Issues Resolved
Contributes to #546 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/Span.java [hunk 1]
```diff
@@ -132,4 +132,11 @@
      * @since 1.2
      */
     TraceGroupFields getTraceGroupFields();
+
+    /**
+     * Gets the serviceName of this span.
+     * @return the ServiceName
+     * @since 1.2
```
Comment (cmanning09) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/Span.java:139]: 1.3

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/JacksonSpan.java [hunk 2]
```diff
@@ -357,6 +363,43 @@ public Builder withTraceGroupFields(final TraceGroupFields traceGroupFields) {
...
+         */
+        public Builder withServiceName(final String serviceName) {
+            data.put(SERVICE_NAME_KEY, serviceName);
+            return this;
+        }
+
+        /**
+         * Sets all attributes by copying over those from another span
+         * @param span
+         * @since 1.2
```
Comment (cmanning09) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/JacksonSpan.java:379]: 1.3

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/JacksonSpan.java [hunk 3]
```diff
@@ -357,6 +363,43 @@ public Builder withTraceGroupFields(final TraceGroupFields traceGroupFields) {
             return this;
         }
 
+        /**
+         * Sets the service name of the span
+         * @param serviceName
+         * @since 1.2
```
Comment (cmanning09) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/JacksonSpan.java:369]: 1.3

**Review by cmanning09 — COMMENTED** [review]
I have a few minor pieces of feedback around javadocs. Otherwise, it looks good.

**Review by cmanning09 — APPROVED** [review]

**Review by graytaylor0 — APPROVED** [review]

**Pull request merged**